### PR TITLE
Add Covariate Shift 

### PR DESCRIPTION
**Pull request by rasensio1**
Improved performance from script in whizzml/examples/scripts.

### CODE REVIEW

**Comment by whizzmler** [comment]
> Should I just wait until that has been done to make the PR?

yes

**Comment by rasensio1** [comment]
All done, ready for review.

File: covariate-shift/covariate-shift.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    "name": "Covariate Shift",
+    "description": "Indicates whether there is a data distribution shift between two datasets. Tries to predict whether a data point came from a 'Training' dataset or 'Production' dataset.",
+    "inputs": [
+        {
+            "name": "train-dst",
```
Comment (jaor) [inline covariate-shift/covariate-shift.json]: i would use `training-dataset` and `production-dataset`, for clarity

File: covariate-shift/covariate-shift.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        }
+    ],
+    "outputs": [
+        {
+            "name": "output",
+            "type": "map",
+            "description": "Map of results. Displays both the average of all individual phi coeffiencet runs and the average across all runs."
+        }
+    ]
+}
```
Comment (jaor) [inline covariate-shift/covariate-shift.json]: this file is miising "kind" and "source_code", and be called "metadata.json".  likewise, the script should be named "script.whizzml".  see any other script in this collection for examples.

File: covariate-shift/covariate-shift.whizzml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+               (append out (sample-and-score
+                             comb-data
+                             (str "test-" (head seeds)))))))))
+
+(define output (multi-phis n train-dst prod-dst))
+
+;; EXAMPLE
+;;
+;; (multi-phis 3
+;;    "dataset/56c3af3f7e0a8d6cca01292e""dataset/56c3af3f7e0a8d6cca01292e")
```
Comment (jaor) [inline covariate-shift/covariate-shift.whizzml]: this example suggests that using the same dataset for train and prod is a good idea... is it?

File: covariate-shift/covariate-shift.json (hunk 2 above)
Comment (jaor) [inline covariate-shift/covariate-shift.json]: also, let's add an entry to the list of examples in the global readme file

File: covariate-shift/covariate-shift.whizzml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+(define (train-data dst-id)
+  (with-origin-field dst-id  "Training"))
+
+(define (prod-data dst-id)
+  (with-origin-field dst-id "Production"))
+
+;; combined-data
+;;
+;; combines two datasets into one new dataset
```
Comment (jaor) [inline covariate-shift/covariate-shift.whizzml]: Combines

File: covariate-shift/covariate-shift.whizzml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+;;                      that holds the values "Production"
+;; Output: (string) Dataset ID for new dataset
+
+(define (combined-data train-dst prod-dst)
+  (create-and-wait-dataset {"origin_datasets" [(train-data train-dst)
+                                               (prod-data prod-dst)]}))
+
+;; sample-dataset
+;;
+;; samples a dataset using bagging. This lets you split one dataset into two.
```
Comment (jaor) [inline covariate-shift/covariate-shift.whizzml]: Samples

File: covariate-shift/covariate-shift.whizzml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+;; Inputs:
+;;  model-id: (string) Model ID
+;;  dst-id: (string) Dataset ID
+
+(define (model-evaluation model-id dst-id)
+  (create-and-wait-evaluation {"model" model-id "dataset" dst-id}))
+
+;; avg-phi
+;;
+;; gets the property "average_phi" from the given evaluation
```
Comment (jaor) [inline covariate-shift/covariate-shift.whizzml]: Gets

File: covariate-shift/covariate-shift.json (hunk 2 above)
Comment (rasensio1) [inline covariate-shift/covariate-shift.json]: Ok. There was at least one in the repo that this structure, but I will change this as you suggest.

File: covariate-shift/covariate-shift.whizzml (hunk 3 above)
Comment (rasensio1) [inline covariate-shift/covariate-shift.whizzml]: No, it's not. I'll change that.

**Comment by rasensio1** [comment]
Comments addressed.

**Comment by jaor** [comment]
Almost there: there should be an entry for this script in the readme.md at the root of the repository

**Comment by rasensio1** [comment]
Got it.

File: covariate-shift/script.whizzml [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    (loop (seeds (range 0 n) out [])
+      (if (= [] seeds)
+        {"list" out
+         "average" (/ (reduce + 0 out) (count out))}
+        (recur (tail seeds)
+               (append out (sample-and-score
+                             comb-data
+                             (str "test-" (head seeds)))))))))
+
+(define output (multi-phis n train-dst prod-dst))
```
Comment (jaor) [inline covariate-shift/script.whizzml]: well, besides changing the names in the metadata, you have to change them here too

**Comment by jaor** [comment]
one more thing...

File: covariate-shift/covariate-shift.json (hunk 2 above)
Comment (jaor) [inline covariate-shift/covariate-shift.json]: oh, which one? we'll change it...

**Comment by rasensio1** [comment]
Okdoke!

**Pull request merged**